Include/internal/pycore_runtime.h

```diff
@@ -118,25 +120,32 @@ typedef struct pyruntimestate {
 
     struct _Py_unicode_runtime_ids unicode_ids;
 
-    struct _Py_global_objects global_objects;
-    // If anything gets added after global_objects then
-    // _PyRuntimeState_reset() needs to get updated to clear it.
+    struct {
+        // The fields here are values that would otherwise have been
+        // malloc'ed during runtime init in pystate.c and pylifecycle.c.
+        // This allows us to avoid allocation costs during startup and
+        // helps simplify the startup code.
+        struct _Py_global_objects global_objects;
+        struct _is interpreters_main;
```

Comment:
`main_interpreter` would be better.

Reply:
The point was to match the relevant field, `_PyRuntimeState.interpreters.main`.  I figured it would make it a little more obvious what this field was for.  I don't mind using the more readable name, if it matters.  Just keep in mind that fields in `_preallocated` will only be accessed directly during init.